blogs/management/commands/update_blogs.py

```diff
@@ -13,24 +13,14 @@ def handle(self, **options):
             entries = get_all_entries(feed.feed_url)
 
             for entry in entries:
-                try:
-                    e = BlogEntry.objects.get(url=entry['url'])
-
-                    # Update our info if it's changed
-                    if e.pub_date < entry['pub_date']:
-                        e.title = entry['title']
-                        e.summary = entry['summary']
-                        e.pub_date = entry['pub_date']
-                        e.save()
-
-                except BlogEntry.DoesNotExist:
-                    BlogEntry.objects.create(
-                        title=entry['title'],
-                        summary=entry['summary'],
-                        pub_date=entry['pub_date'],
-                        url=entry['url'],
-                        feed=feed,
-                    )
+                url = entry.pop('url')
+                e, created = BlogEntry.objects.get_or_create(
+                    feed=feed, url=url, defaults=entry
+                )
+
+                # Update our info if it's changed
+                if not created and e.pub_date < entry['pub_date']:
```

Comment:
This `pub_date` check prevented me from just using `update_or_create`. I'm not sure if this check actually adds much, but I didn't want to mess with the logic too much.

Reply:
I think `e.pub_date < entry['pub_date']` was used because `get_or_create()` wasn't added to Django (just a guess, of course) at the time of writing. We can safely remove that check now. `created` should cover the same use case.